src/DefaultEmissionManager.sol

```diff
@@ -68,10 +68,12 @@ contract DefaultEmissionManager is Ownable2StepUpgradeable, IDefaultEmissionMana
         emit TokenMint(amountToMint, msg.sender);
 
         IPolygonEcosystemToken _token = token;
+        
         _token.mint(address(this), amountToMint);
+
         _token.safeTransfer(treasury, treasuryAmt);
-        // backconvert POL to MATIC before sending to StakeManager
-        migration.unmigrateTo(stakeManager, stakeManagerAmt);
+
+        _token.safeTransfer(stakeManager, stakeManagerAmt);
```

Comment:
Despite this will solve potential issues while unmigrating during minting (if migration contract does not have enough MATIC) it will eventually create an important issue with rewards allocation at the StakeManager contract due to surplus MATIC will be underwater.  Here the script it is being used to keep track of surplus along the way https://github.com/Polypheral/rewards-toolkit/blob/main/scripts/stakingDashboardV2.js#L24